otter/integration/lib/autoscale.py

```diff
@@ -315,27 +330,99 @@ def poll():
             deferred_description=report,
         )
 
+    def wait_for_expected_state(self, _, rcs, timeout=60, period=1,
+                                active=None, pending=None, desired=None):
+        """
+        Repeatedly get the group state until either the specified timeout has
+        occurred or the specified number of active, pending, and desired
+        servers is observed. Unspecifed quantities default to None and are
+        treated as don't cares.
+        """
+
+        def check((code, response)):
+            if code != 200:
+                raise BreakLoopException(
+                    "Could not get the scaling group state"
+                )
+
+            n_active = len(response["group"]["active"])
+            n_pending = response["group"]["pendingCapacity"]
+            n_desired = response["group"]["desiredCapacity"]
+
+            if ((active is None or active == n_active)
```

Comment:
NON-BLOCKING --- I'm wondering if writing `(active in [None, n_active])` (and similar for `pending` and `desired`) would mean the same thing?  Something to consider for a future PR.